soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -7095,16 +7105,39 @@ s32 Player_TryLeavingCrawlspace(Player* this, PlayState* play) {
 
             if (this->linearVelocity > 0.0f) {
                 this->actor.shape.rot.y = this->actor.wallYaw + 0x8000;
-                Player_AnimPlayOnce(play, this, &gPlayerAnim_link_child_tunnel_end);
-                Player_AnimReplaceApplyFlags(play, this, 0x9D);
-                OnePointCutscene_Init(play, 9601, 999, NULL, MAIN_CAM);
+                // #region SOH [Enhancement]
+                if (CVarGetInteger("gCrawlSpeed", 1) > 1) {
+                    // animation when exiting a tunnel forward
+                    LinkAnimation_Change(play, &this->skelAnime, &gPlayerAnim_link_child_tunnel_end,
+                                         1.0f * ((CVarGetInteger("gCrawlSpeed", 1) + 1.0f) / 2.0f), 0.0f,
+                                         Animation_GetLastFrame(&gPlayerAnim_link_child_tunnel_end), ANIMMODE_ONCE,
+                                         0.0f);
+                    Player_AnimReplaceApplyFlags(play, this, 0x9D);
+                    OnePointCutscene_Init(play, 9601, 999, NULL, MAIN_CAM);
+                // #endregion
+                } else {
+                    Player_AnimPlayOnce(play, this, &gPlayerAnim_link_child_tunnel_end);
+                    Player_AnimReplaceApplyFlags(play, this, 0x9D);
+                    OnePointCutscene_Init(play, 9601, 999, NULL, MAIN_CAM);
+                }
             } else {
                 this->actor.shape.rot.y = this->actor.wallYaw;
-                LinkAnimation_Change(play, &this->skelAnime, &gPlayerAnim_link_child_tunnel_start, -1.0f,
-                                     Animation_GetLastFrame(&gPlayerAnim_link_child_tunnel_start), 0.0f, ANIMMODE_ONCE,
-                                     0.0f);
-                Player_AnimReplaceApplyFlags(play, this, 0x9D);
-                OnePointCutscene_Init(play, 9602, 999, NULL, MAIN_CAM);
+                // #region SOH [Enhancement]
+                // animation when exiting a tunnel backward 
+                if (CVarGetInteger("gCrawlSpeed",1) > 1) {
+                    LinkAnimation_Change(play, &this->skelAnime, &gPlayerAnim_link_child_tunnel_start,
+                                         -1.0f * ((CVarGetInteger("gCrawlSpeed", 1) + 1.0f) / 2.0f),
```

Comment:
Why the `+ 1.0f) / 2.0f`?

Would be simpler with:
```suggestion
                                         -1.0f * (CVarGetInteger("gCrawlSpeed", 1)),
```